l1-contracts/src/governance/GSE.sol

```diff
@@ -306,10 +306,6 @@ contract GSECore is IGSECore, Ownable {
     bool _moveWithLatestRollup
   ) external override(IGSECore) onlyRollup {
     if (checkProofOfPossession) {
```

Comment:
Oh, I meant moving the entire check of proof of possession (the full if) further down. 

But I guess that the check for previous will actually catch the same as the `isRegisted` would. There is however still the failure cases when `_moveWithLatest` rollup etc can make it fail, which seems a fairly painful way to revert as you would still have paid for the pairings and all that, just to fail from a read that could have been done before.